frappe/email/smtp.py

```diff
@@ -171,7 +171,7 @@ def setup_email_account(self, append_to=None, sender=None):
 		if self.email_account:
 			self.server = self.email_account.smtp_server
 			self.login = getattr(self.email_account, "login_id", None) or self.email_account.email_id
-			self.password = self.email_account.password
+			self.password = str(self.email_account.password)
```

Comment:
use `frappe.safe_decode` so it will work in both 2 and 3

Reply:
How about something like this instead
```
			if self.email_account.password.__class__.__name__ == 'unicode':
				try:
					self.password = str(self.email_account.password)
				except UnicodeEncodeError:
					self.password = frappe.safe_decode(self.email_account.password)
```

Reply:
@ioweb-gr `str` will also return a unicode value

can you try `frappe.safe_encode` ?

Reply:
@rmehta I tried it with `frappe.safe_encode(param, 'ascii')` and it worked for me. What will happen if the user actually has a unicode character in the password?